e2e/catalog_source_test.go

```diff
@@ -0,0 +1,111 @@
+package e2e_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	k8sApiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	"github.com/openshift/addon-operator/e2e"
+)
+
+func TestAddon_CatalogSource(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	addon := &addonsv1alpha1.Addon{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "addon-oisafbo12",
+			Namespace: "default",
+		},
+		Spec: addonsv1alpha1.AddonSpec{
+			Namespaces: []addonsv1alpha1.AddonNamespace{
+				{Name: "namespace-onbgdions"},
+				{Name: "namespace-pioghfndb"},
+			},
+			Install: addonsv1alpha1.AddonInstallSpec{
+				Type: addonsv1alpha1.OwnNamespace,
+				OwnNamespace: &addonsv1alpha1.AddonInstallOwnNamespace{
+					AddonInstallCommon: addonsv1alpha1.AddonInstallCommon{
+						Namespace:          "namespace-onbgdions",
+						CatalogSourceImage: testCatalogSourceImage,
+					},
+				},
+			},
+		},
+	}
+
+	err := e2e.Client.Create(ctx, addon)
+	require.NoError(t, err)
+
+	// clean up addon resource in case it
+	// was leaked because of a failed test
+	wasAlreadyDeleted := false
```

Comment:
I would rather add a `if err != nil && !errors.IsNotFound(err)` check to the delete error check, instead of remembering if a delete was already performed.

Testing the teardown is already done with `e2e.WaitToBeGone` and I don't see a reason for extra complexity here.

Reply:
Gotcha, thank you :)

Reply:
Done :)